lgn-messages/src/types/v1/query/tasks.rs

```diff
@@ -161,20 +166,146 @@ pub struct SinglePathLeafInput
+pub enum Hydratable<K: Clone + std::fmt::Debug>
+{
+    Dehydrated(K),
+    Hydrated(Arc<Vec<u8>>),
+}
 
-    pub query_proof_location: ProofKey,
+impl<T: Clone + std::fmt::Debug> std::fmt::Debug for Hydratable<T>
+{
+    fn fmt(
+        &self,
+        f: &mut std::fmt::Formatter<'_>,
+    ) -> std::fmt::Result
+    {
+        match self
+        {
+            Hydratable::Dehydrated(k) =>
+            {
+                write!(
+                    f,
+                    "dehydrated: {k:?}"
+                )
+            },
+            Hydratable::Hydrated(_) =>
+            {
+                write!(
+                    f,
+                    "hydrated"
+                )
+            },
+        }
+    }
+}
 
-    #[dbg(placeholder = "...")]
-    pub indexing_proof: Vec<u8>,
+impl<K: Clone + std::fmt::Debug> Hydratable<K>
+{
+    /// Wrap a proof key into a `Dehydrated` variant.
+    pub fn new(k: K) -> Self
+    {
+        Hydratable::Dehydrated(k)
+    }
 
-    #[dbg(placeholder = "...")]
-    pub query_proof: Vec<u8>,
+    /// Consume a `Hydrated` variant into its embedded proof; panic if it is
+    /// not hydrated.
+    pub fn proof(&self) -> Arc<Vec<u8>>
+    {
+        match self
+        {
+            Hydratable::Dehydrated(_) => unreachable!(),
+            Hydratable::Hydrated(proof) => proof.clone(),
+        }
+    }
+
+    /// Consume a `Hydrated` variant into its embedded proof; panic if it is
+    /// not hydrated.
+    pub fn clone_proof(&self) -> Vec<u8>
+    {
+        match self
+        {
+            Hydratable::Dehydrated(_) => unreachable!(),
+            Hydratable::Hydrated(proof) =>
+            {
+                proof
+                    .clone()
+                    .to_vec()
+            },
+        }
+    }
+
+    /// Convert a `Dehydrated` variant into its embedded key; panic if it is
+    /// not hydrated.
+    pub fn key(&self) -> K
+    {
+        match self
+        {
+            Hydratable::Dehydrated(k) => k.clone(),
+            Hydratable::Hydrated(_) => unreachable!(),
+        }
+    }
+
+    /// Hydrates a `Dehydrated` variant; panic if it is already hydrated.
+    pub fn hydrate(
+        &mut self,
+        proof: Vec<u8>,
+    )
+    {
+        assert!(
+            matches!(
+                self,
+                Hydratable::Dehydrated(_)
+            )
+        );
+        *self = Hydratable::Hydrated(Arc::new(proof))
+    }
+}
+
+#[derive(Clone, Dbg, Deserialize, Serialize)]
+pub enum RevelationInput
+{
+    Aggregated
+    {
+        placeholders: PlaceHolderLgn,
+
+        #[dbg(placeholder = "...")]
+        indexing_proof: Hydratable<db_keys::ProofKey>,
+
+        #[dbg(placeholder = "...")]
+        query_proof: Hydratable<ProofKey>,
+    },
+    Tabular
+    {
+        placeholders: PlaceHolderLgn,
+        indexing_proof: Hydratable<db_keys::ProofKey>,
+        matching_rows: Vec<HydratableMatchingRow>,
+        column_ids: ColumnIDs,
+        limit: u32,
+        offset: u32,
```

Comment:
why these are not option ?

Reply:
Because we read them from the assembled circuits, where parsil will already have set them to their default values if they have not been specified by the user.